subprojects/plugins/src/integTest/resources/org/gradle/groovy/compile/BasicGroovyCompilerIntegrationSpec/compileWithAnnotationProcessor/settings.gradle

```diff
@@ -0,0 +1,7 @@
+include ':'
+include ':SimpleAnnotationProcessor'
+// Integration tests don't copy directories that aren't named after a test to build/tmp/test files, where these build scripts are actually run from.
```

Comment:
Looks like I was mistaken when I wrote this comment.

When I looked at the source of [TestResources.java](https://github.com/gradle/gradle/blob/master/subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/TestResources.java#L56) I noticed that a shared directory is also copied with each test.

So I was able to dramatically simplify this logic by moving SimpleAnnotationProcessor into the shared directory, and then that directory is copied into the temporary directory to actually run the integration test.

Considering that the settings.gradle now looks like:
        include ':'
        include ':SimpleAnnotationProcessor'

Do you think it's still worth constructing these inline, within the test?